### PR TITLE
404 value is stored in cache after software release is created.

### PR DESCRIPTION
**Pull request by lomasgupta**
Ticket No: https://github.com/openscholar/openscholar/issues/13043
https://github.com/openscholar/openscholar/issues/13043#issuecomment-616382687

### CODE REVIEW

File: src/Routing/PurlRouteProvider.php [hunk 1]
```diff
@@ -163,7 +163,9 @@ public function getRouteCollectionForRequest(Request $request) {
         'query' => $query_parameters,
         'routes' => $routes,
       ];
-      $this->cache->set($cid, $cache_value, CacheBackendInterface::CACHE_PERMANENT, ['route_match']);
+      if ($routes->all() || $routes->getResources()) {
```
Comment (subhojit777) [inline src/Routing/PurlRouteProvider.php]: Can't we just use one of them? either `all` or `getResources`?

Comment (lomasgupta) [reply src/Routing/PurlRouteProvider.php]: It is working fine with getResources(). I wanted to test it on dev, so added both. Removing all().

File: src/Routing/PurlRouteProvider.php [hunk 2]
```diff
@@ -163,7 +163,9 @@ public function getRouteCollectionForRequest(Request $request) {
         'query' => $query_parameters,
         'routes' => $routes,
       ];
-      $this->cache->set($cid, $cache_value, CacheBackendInterface::CACHE_PERMANENT, ['route_match']);
+      if ($routes->getResources()) {
```
Comment (subhojit777) [inline src/Routing/PurlRouteProvider.php:166]: btw did you check why `getRoutesByPath` is returning incorrect route definitions in the first place?

Comment (lomasgupta) [reply src/Routing/PurlRouteProvider.php:166]: Its returning 2 routes "route:/test-subhojit/test-pro-drupal" and route:/test-subhojit/system/404
Not sure why.

**Review by rbran100 — APPROVED** [review]

**Pull request merged**